### PR TITLE
feat(logger): add `filters` option

### PR DESCRIPTION
**Pull request by lucasfernog**
# Description of change

Adds filter support to the bee-common logger. 

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

With my own logger usage, with and without the filters array.

## Change checklist

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that new and existing unit tests pass locally with my changes
- [x] I have updated the CHANGELOG.md, if my changes are significant enough

### CODE REVIEW

File: bee-common/bee-common/src/logger/config.rs [hunk 1]
```diff
@@ -58,6 +70,8 @@ pub struct LoggerOutputConfig {
     pub(crate) name: String,
     /// Log level of an output.
     pub(crate) level: LevelFilter,
+    /// Log filters of the output.
+    pub(crate) filters: Option<Vec<String>>,
```
Comment (Alex6323) [inline bee-common/bee-common/src/logger/config.rs]: IMHO with another way of filtering the log output we should refactor the naming a bit. I'ld rename `level` to `level_filter` and `filters` to `target_filters`. I think I'ld also introduce a typedef `type TargetFilters = Vec<String>`, but you can ignore this if you don't like to do this. I'ld also rename all the fields accordingly. They are not that short anymore, but I think it helps with clarity.

**Review by Alex6323 — COMMENTED** [review]
Good and helpful PR. Once merged I would update Bee's CLI so that we can easily make use of this added filter option.

**Review by Alex6323 — DISMISSED** [review]

File: bee-common/bee-common/src/logger/config.rs [hunk 2]
```diff
@@ -57,7 +70,9 @@ pub struct LoggerOutputConfig {
     /// Name of an output file, or `stdout` for standard output.
     pub(crate) name: String,
     /// Log level of an output.
-    pub(crate) level: LevelFilter,
+    pub(crate) level_filter: LevelFilter,
+    /// Log filters of the output.
+    pub(crate) target_filters: Option<Vec<String>>,
```
Comment (thibault-martinez) [inline bee-common/bee-common/src/logger/config.rs]: `Option` is needed for the builder to inform serde that the field is optional, but IMO in the built conf we should just have a vector, possibly empty. Option makes it more complex than it should be.

File: bee-common/bee-common/src/logger/config.rs [hunk 3]
```diff
@@ -21,7 +21,9 @@ pub struct LoggerOutputConfigBuilder {
     /// Name of an output file, or `stdout` for standard output.
     name: Option<String>,
     /// Log level of an output.
```
Comment (thibault-martinez) [inline bee-common/bee-common/src/logger/config.rs]:
```suggestion
    /// Log level filter of an output.
```

File: bee-common/bee-common/src/logger/config.rs [hunk 4]
```diff
@@ -21,7 +21,9 @@ pub struct LoggerOutputConfigBuilder {
...
-    level: Option<LevelFilter>,
+    level_filter: Option<LevelFilter>,
+    /// Log filters of the output.
```
Comment (thibault-martinez) [inline bee-common/bee-common/src/logger/config.rs]:
```suggestion
    /// Log target filters of an output.
```

**Review by thibault-martinez — COMMENTED** [review]
Nice feature!

**Review by thibault-martinez — APPROVED** [review]

**Review by Alex6323 — APPROVED** [review]

**Review by samuel-rufi — APPROVED** [review]

**Pull request merged**